port/unix/omrsl.c

```diff
@@ -438,3 +439,71 @@ omrsl_startup(struct OMRPortLibrary *portLibrary)
 {
 	return 0;
 }
+
+/**
+ * Native Libraries
+ *
+ * This function is called go get all the shared libraries loaded by a process.
+ *
+ * @param[in] portLibrary Pointer to the OMR port library.
+ * @param[in] callback Function to be called for each library.
+ * @param[in] userData User-defined data passed to the callback.
+ *
+ * @return 0 if successful, or the first non-zero return value from the callback.
+ */
+uintptr_t
+omrsl_get_libraries(struct OMRPortLibrary *portLibrary, OMRLibraryInfoCallback callback, void *userData)
+{
+#if defined(LINUX)
+/* Length of buffer. */
+#define READ_CHUNK_SIZE (PATH_MAX + 100)
+	char buffer[READ_CHUNK_SIZE + 1];
+	uintptr_t result = 0;
+	void *addrLow = NULL;
+	void *addrHigh = NULL;
+	unsigned long offset = 0;
+	int devMajor = 0;
+	int devMinor = 0;
+	unsigned long inode = 0;
+	char permissions[5] = {0};
+	char path[READ_CHUNK_SIZE];
+	int count = 0;
+	int32_t portableError = 0;
+	intptr_t fd = portLibrary->file_open(portLibrary, "/proc/self/maps", EsOpenRead, 0);
+	if (-1 == fd) {
+		portableError = portLibrary->error_last_error_number(portLibrary);
+		Trc_PRT_failed_to_open_proc_maps(portableError);
+		portLibrary->error_set_last_error_with_message(
+				portLibrary,
+				portableError,
+				"Failed to open /proc/self/maps");
+		return (uintptr_t)(intptr_t)portableError;
+	}
+	while (NULL != portLibrary->file_read_text(portLibrary, fd, buffer, sizeof(buffer))) {
+		addrLow = NULL;
+		addrHigh = NULL;
+		offset = 0;
+		devMajor = 0;
+		devMinor = 0;
+		inode = 0;
+		memset(permissions, 0, sizeof(permissions));
+		memset(path, 0, sizeof(path));
+		count = sscanf(
+				buffer,
+				"%p-%p %4s %lx %x:%x %lu %s",
```

Comment:
Suggest that the local `path` is not needed; instead of `%s` for the final pattern, use `%n` to learn the offset within `buffer` where the path is located.